helpers/response/query_response.py

```diff
@@ -0,0 +1,90 @@
+from datetime import datetime
+from graphql import GraphQLError
+from utilities.validations import validate_date_range
+
+
+def filter_rooms_by_responses(
+    Query, info, upper_limit_count, lower_limit_count
+):
+    all_responses = Query().get_all_reponses(info)
+    filtered_responses = []
+    for response in all_responses:
+        reponse_count = response.total_responses
+        if lower_limit_count <= reponse_count <= upper_limit_count:
+            filtered_responses.append(response)
+    return filtered_responses
+
+
+def filter_response_by_date(Query, info, end_date, start_date):
+    all_responses = Query().get_all_reponses(info)
+    filtered_responses = []
+    for responses in all_responses:
+        for response in responses.response:
+            validate_response_dates(
+                response,
+                end_date, start_date, filtered_responses
+            )
+        setattr(responses, 'response', filtered_responses)
+        filtered_responses = []
+    return all_responses
+
+
+def validate_response_dates(
+    response, end_date, start_date, filtered_responses
+):
+    end_date = datetime.strptime(end_date, '%Y %b %d')
+    start_date = datetime.strptime(start_date, '%Y %b %d')
+    validate_date_range(
+        end_date=end_date, start_date=start_date
+    )
+    if (response.created_date < end_date and
+            response.created_date > start_date):
+        filtered_responses.append(response)
+    return filtered_responses
+
+
+def check_limits_are_provided(lower_limit, upper_limit, typ):
```

Comment:
what does `typ` variable represent. Can you use a more descriptive name.

Reply:
Thanks, @verenceLola, this has been changed.